pkg/virt-launcher/virtwrap/nichotplug.go

```diff
@@ -120,3 +124,41 @@ func indexedDomainInterfaces(domain *api.Domain) map[string]api.Interface {
 	}
 	return domainInterfaces
 }
+
+// withNetworkIfacesResources adds network interfaces as placeholders to the domain spec
+// to trigger the addition of the dependent resources/devices (e.g. PCIe).
+// As its last step, it reads the generated configuration and removed the network interfaces
+// so none will be created with the domain creation.
+// The dependent devices are left in the configuration, to allow future hotplug.
+func withNetworkIfacesResources(vmi *v1.VirtualMachineInstance, domainSpec *api.DomainSpec, f func(v *v1.VirtualMachineInstance, s *api.DomainSpec) (cli.VirDomain, error)) (cli.VirDomain, error) {
+	domainSpecWithIfacesResource := domainSpec.DeepCopy()
+	requests := resource.ExtendedResourceList{ResourceList: vmi.Spec.Domain.Resources.Requests}
+	reqInterfaces := int(requests.Interface().Value())
+	for i := 0; i < (reqInterfaces - len(vmi.Spec.Domain.Devices.Interfaces)); i++ {
+		domainSpecWithIfacesResource.Devices.Interfaces = append(domainSpecWithIfacesResource.Devices.Interfaces, api.Interface{
+			Type:  "ethernet",
+			Model: &api.Model{Type: "virtio-non-transitional"},
```

Comment:
afaiu, you cannot use directly this type - it must be calculated.

Are you re-calculating its value later on ?

Reply:
I think we do not support hotplug for the cases it is not like this.
Maybe I missed something, but in what scenario and what would be the result if used?

Reply:
Thinking of it again, I do not think it has an effect on the dependent resources of the domain, it only has impact on the vNIC itself, and we remove it before creating the domain itself.

Reply:
Hm, but if we remove it, why have it at all ? Is it a mandatory attribute ?

Reply:
Good question, I have not checked it.
@andreabolognani , maybe you know?

Reply:
The attribute is _technically_ not mandatory, but the libvirt default (`rtl8139`) is so poor that, for all intents and purposes, you have to specify it.

AFAICT KubeVirt uses the `virtio-non-transitional` model for its network interfaces by default, at least for q35. If there are different default models for different machine types (I haven't checked), then the default model for the machine type should be used.

This will ensure that the PCI controllers that are automatically added during this phase can actually be used later on by the hotplugged interfaces.

Reply:
[UseVirtioTransitional](https://github.com/kubevirt/kubevirt/blob/24a67ecc90a44cc6109465bec109f216d1af2676/staging/src/kubevirt.io/api/core/v1/schema.go#L405) controls this.

I'm not really sure if we need to have this support.
I tend to defer this to when someone will actually report this as a problem and only then consider adding it.

I do not have strong opinions on this.

Reply:
Yeah, we need to use the same logic here. If we don't, then for `q35` machines with `UseVirtioTransitional` enabled we'll end up allocating a bunch of `pcie-root-port` controllers that will remain unused even after all the hotplug operations.

Reply:
I will look into this, but I have a feeling we are trying to support an edge case which may not worth the effort.
If it is simple, I'll add the logic.
To be continued...

Reply:
The dependency is on `vmi.Spec.Domain.Devices.UseVirtioTransitional` directly.
I added it to the placeholder interface creation.
Done